### PR TITLE
CORE-405 / CORE-417 - FTE Image Uploading Bug Fixes

### PR DESCRIPTION
**Pull request by atoff**
Note, will require existing images to be move from public/storage/smartcars... to public/smartcars... 

Closes CORE-405 and CORE-417

### CODE REVIEW

**Comment by AxonC** [comment]
Tests failing old chap @atoff

File: app/Libraries/Storage/CoreUploadedFile.php [hunk 1]
```diff
@@ -15,4 +15,9 @@ public function getFullFileName()
     {
         return sha1("{$this->getClientOriginalName()}.{$this->getClientOriginalExtension()}");
     }
+
+    public function getFilePathName()
+    {
+        return $this->getFullFileName() . '.' . $this->extension();
```
Comment (AxonC) [inline app/Libraries/Storage/CoreUploadedFile.php]: Consistency with the method above with respect to string construction please me duck

File: app/Libraries/Storage/FteStorageWrapper.php [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 class FteStorageWrapper extends StorageWrapper
 {
-    protected $basePath = "storage/smartcars/exercises";
+    protected $basePath = "smartcars/exercises";
```
Comment (AxonC) [inline app/Libraries/Storage/FteStorageWrapper.php]: Why are we changing the path?

Comment (atoff) [reply app/Libraries/Storage/FteStorageWrapper.php]: Because, to me at least, it doesn't make sense to store in public/storage/storage/smartcars...

Comment (AxonC) [reply app/Libraries/Storage/FteStorageWrapper.php]: I think that was done due to some of the symlinking in prod; where storage/storage corresponded to public/storage

**Comment by atoff** [comment]
> Tests failing old chap @atoff

that would be because I forgot to commit the test update :///

Should be good now

File: app/Libraries/Storage/FteStorageWrapper.php (hunk 2 above)
Comment (atoff) [reply app/Libraries/Storage/FteStorageWrapper.php]: it should be that storage/public maps to public/storage. If its not this is a prod issue not standard practice, which we probably need to look at fixing

**Comment by atoff** [comment]
@AxonC I've merged changes for CORE-417 in the latest commit (a8700ad) - just fixes the deleting aspect aswell

**Pull request merged**